File: 백준/Silver/1920. 수 찾기/수 찾기.js
const fs = require('fs')

const [a1,a2,a3,a4] = fs.readFileSync('/dev/stdin').toString().trim().split("\n");

const dic = {}

for(var s of a2.trim().split(" ")) {
    dic[s] = ""
}

for(var s of a4.trim().split(" ")) {
    console.log((s in dic) ? 1 : 0)
}

File: 백준/Silver/1920. 수 찾기/수 찾기.py
import sys

input = sys.stdin.readline

n = int(input())
arr = [int(i) for i in input().split()]
arr.sort()

m = int(input())
arr2 = [int(i) for i in input().split()]

def main():
    dic = {}
    for i in arr:
        dic[i] = 0
    
    for i in arr2:
        if i in dic:
            print(1)
        else:
            print(0)
            
main()
